docs/content/testing/testing-guidelines.md

```diff
@@ -0,0 +1,238 @@
+
+-----
+
+# Testing guidelines
+
+The tests should:
+
+- Set up some initial state
+- Execute a method that acts upon this state and results in a new state
+- Make assertions in the new state against the expected changes
+- Clean up
+
+NUnit allows this to be done through the use of fixtures and special method annotations.
+
+## Module directory structure
+
+The Unity GDK project's assets are divided into modules, such as `Core`, `Physics`, `PlayerLifecycle`.
+
+Each module is in its own directory within `workers\unity\Assets\Gdk`.
+
+Each module has its own tests that cover the functionality of that module.
+
+Unity Test Runner also allows separate Playmode and Editmode tests. For more information on this division, please refer to the [Unity Test Runner manual page](https://docs.unity3d.com/Manual/testing-editortestsrunner.html).
+
+Within the root of the module, the `Tests` directory should contain `Editmode`
+ and `Playmode` subdirectories with their own assembly definition files.
+
+Test assembly definition files must match this pattern:
+> `Improbable.Gdk.<ModuleName>.<TestMode>Tests.asmdef`
+
+### Editmode vs. Playmode tests
+
+As many unit tests as possible should be Editmode tests. This is because Editmode tests are faster, since they do not require scenes and Gameobjects to be set up.
+
+Playmode-specific features should be mocked as much as possible, unless it’s impractical to do so. For example, if a function depends on some time passing, it should be using a mockable time interface instead of the `UnityEngine.Time` class directly.
+
+If it is impossible to do a test as an editor test, e.g. a particular kind of functionality that responds to a physics callback, then they can be Playmode tests.
+
+Another justification for a Playmode test is to check that certain functionality will still be present when a game is being played, or for more complicated integration testing that may depend on multiple systems interacting with each other.
+
+### Test Folder Structure
+
+Within each test folder, the folder structure should match the non-test code structure.
+
+If any class file path can be broken down into the pattern `<module path>/<path within module>.cs`, the Editmode tests for this class should be in `<module path>/Tests/Editmode/<path within module>Tests.cs`.
+
+For a more concrete example, for the class  `workers/unity/Assets/Gdk/Core/Utility/CommandLineUtility.cs` the test fixtures should be within `workers/unity/Assets/Gdk/Core/Tests/Editmode/Utility/CommandLineUtilityTests.cs`.
+
+If a class has both Editmode and Playmode tests, you can name the test filenames as `<ClassName>EditmodeTests.cs` and `<ClassName>PlaymodeTests.cs` within their respective assemblies.
+
+## File structure
+
+### Namespaces
+
+The tests should be grouped under this namespace pattern:
+
+`Improbable.Gdk.<ModuleName>.<TestMode>Tests`
+
+Where:
+- `<ModuleName>` is the GDK module being tested (e.g. `Core`)
+- `<TestMode>` is one of `Editmode` or `Playmode`
```

Comment:
`EditMode` and `PlayMode`?

Reply:
Unity uses `Editmode`, I will change `TestMode` to be consistent

Reply:
Actually I just checked again and I'm wrong, will fix